shiny/_main.py

```diff
@@ -261,10 +284,12 @@ def run_app(
 
     if reload_dirs is None:
         reload_dirs = []
+        if app_dir is not None:
+            reload_dirs = [app_dir]
 
     if reload:
         # Always watch the app_dir
-        if app_dir:
+        if app_dir and app_dir not in reload_dirs:
             reload_dirs.append(app_dir)
```

Comment:
`app_dir` is added to `reload_dirs` above iff `reload_dirs` is its default value and `app_dir` is supplied.
So, we should remove this enhancement.

```suggestion
```